custom/icds_reports/utils/aggregation_helpers/distributed/agg_service_delivery_report.py

```diff
@@ -0,0 +1,256 @@
+from dateutil.relativedelta import relativedelta
+
+from corehq.apps.userreports.util import get_table_name
+from custom.icds_reports.const import AGG_SDR_TABLE
+from custom.icds_reports.utils.aggregation_helpers import  month_formatter
+from custom.icds_reports.utils.aggregation_helpers.distributed.base import AggregationPartitionedHelper
+
+
+class AggServiceDeliveryReportHelper(AggregationPartitionedHelper):
+    helper_key = 'agg-service_delivery_report'
+    base_tablename = AGG_SDR_TABLE
+    staging_tablename = 'staging_{}'.format(AGG_SDR_TABLE)
+
+    @property
+    def monthly_tablename(self):
+        return "{}_{}".format(self.base_tablename, self.month.strftime("%Y-%m-%d"))
+
+    @property
+    def previous_agg_table_name(self):
+        return f"previous_{self.monthly_tablename}"
+
+    @property
+    def model(self):
+        from custom.icds_reports.models.aggregate import AggServiceDeliveryReport
+        return AggServiceDeliveryReport
+
+    def update_queries(self):
```

Comment:
These would be a lot faster if the temp tables are are unlogged and if they are distributed. You can see how agg_awc or agg_child_health does this